the-blue-alliance-ios/View Controllers/Teams/TeamSummaryViewController.swift

```diff
@@ -352,21 +343,18 @@ extension TeamSummaryViewController: Refreshable {
 
         // Refresh awards
         var awardsRequest: URLSessionDataTask?
-        awardsRequest = TBAKit.sharedKit.fetchTeamAwards(key: team.key!, eventKey: event.key!, completion: { (awards, error) in
+        awardsRequest = TBAKit.sharedKit.fetchTeamAwards(key: teamKey.key!, eventKey: event.key!, completion: { (awards, error) in
             if let error = error {
-                self.showErrorAlert(with: "Unable to refresh event awards for \(self.team.key!) - \(error.localizedDescription)")
+                self.showErrorAlert(with: "Unable to refresh event awards for \(self.teamKey.key!) - \(error.localizedDescription)")
             }
 
             self.persistentContainer.performBackgroundTask({ (backgroundContext) in
-                let backgroundEvent = backgroundContext.object(with: self.event.objectID) as! Event
                 if let awards = awards {
-                    let localAwards = awards.map({ (modelAward) -> Award in
-                        return Award.insert(with: modelAward, for: backgroundEvent, in: backgroundContext)
-                    })
-                    backgroundEvent.awards = Set(localAwards) as NSSet
+                    Award.insert(awards, event: self.event, in: backgroundContext)
```

Comment:
TODO: This needs to have a Event on the proper context to setup relationships properly